tests/test-files/define-type-alias.txt

```diff
@@ -0,0 +1,142 @@
+================================================================================
+1 Define type alias
+================================================================================
+
+(package coalton-unit-tests)
+
+(define-type-alias Index Integer)
+
+================================================================================
+2 Define type alias
+================================================================================
+
+(package coalton-unit-tests)
+
+(define-type-alias UnaryIntegerOperator (Integer -> Integer))
+
+================================================================================
+3 Define type alias
+================================================================================
+
+(package coalton-unit-tests)
+
+(define-type-alias (UnaryOperator :a) (:a -> :a))
+
+================================================================================
+4 Define type alias
+================================================================================
+
+(package coalton-unit-tests)
+
+(define-type-alias (ReverseTranslationRules :a :b) (:b -> :a))
```

Comment:
I believe I found a bug when there are multiple type variables in the alias. It seems to only detect the first argument passed in. Here is the test code:

```lisp
(coalton-toplevel
  (define-type-alias (MyTuple :a :b) (Tuple :a (Tuple Integer :b)))

  (declare sum-mytup ((MyTuple Integer Integer) -> Integer))
  (define (sum-mytup (Tuple a (Tuple b c))) (+ a b c)))
```
Here is the error that I am getting:
```
(during macroexpansion of (COALTON-TOPLEVEL
  (DEFINE-TYPE-ALIAS #
    ...)
  ...))
error: Incomplete type alias application
  --> <macroexpansion>:4:21
   |
 4 |    (DECLARE SUM-MYTUP ((MYTUPLE INTEGER INTEGER) -> INTEGER))
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Type alias MYTUPLE is applied to 1 argument, but 2 arguments are required.
```

In the test here, it looks like an alias for a type with multiple variables is defined, but not actually used.

Reply:
Thank you for pointing this out, I believe I know what the problem is, and I will try to address it tomorrow.

Reply:
This bug should be fixed with my latest commit. I will clean up my code, add detailed comments and documentation, and add test cases. I wanted to note that the function `+` takes two arguments, so if anyone wants to try this bug again, make sure to replace `(+ a b c)` with `(+ a (+ b c)` or something similar.